parsec/api/data/message.py

```diff
@@ -46,7 +47,7 @@ def get_obj_type(self, obj: Dict[str, object]) -> object:
 class SharingGrantedMessageContent(BaseMessageContent):
     class SCHEMA_CLS(BaseSignedDataSchema):
         type = fields.EnumCheckedConstant(MessageContentType.SHARING_GRANTED, required=True)
-        name = fields.String(required=True)
+        name = EntryNameField(validate=validate.Length(min=1, max=256), required=True)
```

Comment:
Length is already checked by `EntryName`

btw it check the size of the utf8-encoded string given it correspond to the actual size of stored data, here you check the size of the string which can be different if the string contains non-ascii characters:
```python
>>> len("foo".encode("utf8"))
3
>>> len("ééé".encode("utf8"))
6
>>> "ééé".encode("utf8")
b'\xc3\xa9\xc3\xa9\xc3\xa9'
>>> len("🐍🐍🐍".encode("utf8"))
12
>>> "🐍🐍🐍".encode("utf8")
b'\xef\xbf\xbd\xef\xbf\xbd\xef\xbf\xbd'
```

Reply:
I guess the trouble comes from `WorkspaceEntry`, `FolderManifest` and `WorkspaceManifest`, can you fix them ?